packages/wdio-config/src/node/ConfigParser.ts

```diff
@@ -177,8 +177,18 @@ export default class ConfigParser {
     private merge(object: MergeConfig = {}, addPathToSpecs = true) {
         const spec = Array.isArray(object.spec) ? object.spec : []
         const exclude = Array.isArray(object.exclude) ? object.exclude : []
-        this._config = deepmerge(this._config, object) as TestrunnerOptionsWithParameters
-
+        const customDeepMerge = deepmergeCustom({
```

Comment:
One last ask: can we add a comment describing why we do a custom deepmerge here? It should mention that the default merge would duplicate entries when using the launcher programatically.

Reply:
Sure I'll do